lib/dal/src/workspace_integrations.rs

```diff
@@ -0,0 +1,126 @@
+use crate::{workspace::WorkspaceId, DalContext, TransactionsError};
+use serde::{Deserialize, Serialize};
+use si_data_pg::{PgError, PgRow};
+use thiserror::Error;
+
+#[remain::sorted]
+#[derive(Error, Debug)]
+pub enum WorkspaceIntegrationsError {
+    #[error(transparent)]
+    Pg(#[from] PgError),
+    #[error("transactions error: {0}")]
+    Transactions(#[from] TransactionsError),
+}
+
+pub type WorkspaceIntegrationsResult<T> = Result<T, WorkspaceIntegrationsError>;
+
+pub use si_id::WorkspaceIntegrationId;
```

Comment:
Ironically I kept this pattern purely because I didn't want the original `si-id` PR to update **_every_** call site. Though I can't say I hate if we continue this either rather than just `use si_id::<my-id>`... hmm... at any rate, looks good!